examples/jsm/nodes/shadernode/ShaderNode.js

```diff
@@ -147,31 +148,43 @@ const ShaderNodeImmutable = function ( NodeClass, ...params ) {
 
 };
 
-const ShaderNodeScript = function ( jsFunc ) {
+class ShaderNodeInternal extends Node {
+
+	constructor( jsFunc ) {
 
-	// @TODO: Move this to Node extended class
+		super();
 
-	const self = {
+		this._jsFunc = jsFunc;
 
-		build: ( builder ) => {
+	}
 
-			self.call( {}, builder );
+	call( inputs, builder ) {
 
-			return '';
+		inputs = nodeObjects( inputs );
+
+		return nodeObject( this._jsFunc( inputs, builder ) );
+
+	}
 
-		},
+	build( builder ) {
 
-		call: ( inputs, builder ) => {
+		const output = this.call( {}, builder );
 
-			inputs = nodeObjects( inputs );
+		if ( output === undefined ) {
 
-			return nodeObject( jsFunc( inputs, builder ) );
+			return '';
 
 		}
 
-	};
+		return output.build( builder );
```

Comment:
It seems that this can produce some artifacts like `NodeBuffer_347.nodeUniform0[ instanceIndex ] = vec4<u32>( instanceIndex, instanceIndex, instanceIndex, 1u ); NodeBuffer_347.nodeUniform0[ instanceIndex ];` (repeating left-hand side of an assignment as a separate expression), which do not work in WGSL...